wkcuber/api/layer.py

```diff
@@ -415,27 +415,36 @@ def downsample(
 
         if max_mag is None:
             max_mag = calculate_default_max_mag(
-                self.dataset.properties.data_layers[self.name].get_bounding_box_size()
+                self.dataset.properties.data_layers[self.name].get_bounding_box_size(),
             )
 
-        if sampling_mode == SamplingModes.AUTO:
+        scale: Optional[Tuple[float, float, float]] = None
+        if sampling_mode == SamplingModes.ANISOTROPIC or sampling_mode == "auto":
             scale = self.dataset.properties.scale
         elif sampling_mode == SamplingModes.ISOTROPIC:
-            scale = (1, 1, 1)
+            scale = None
         elif sampling_mode == SamplingModes.CONSTANT_Z:
             max_mag_with_fixed_z = max_mag.to_array()
             max_mag_with_fixed_z[2] = from_mag.to_array()[2]
             max_mag = Mag(max_mag_with_fixed_z)
-            scale = calculate_virtual_scale_for_target_mag(max_mag)
+            scale = self.dataset.properties.scale
         else:
             raise AttributeError(
-                f"Downsampling failed: {sampling_mode} is not a valid SamplingMode ({SamplingModes.AUTO}, {SamplingModes.ISOTROPIC}, {SamplingModes.CONSTANT_Z})"
+                f"Downsampling failed: {sampling_mode} is not a valid SamplingMode ({SamplingModes.ANISOTROPIC}, {SamplingModes.ISOTROPIC}, {SamplingModes.CONSTANT_Z})"
             )
 
-        prev_mag = from_mag
-        target_mag = get_next_mag(prev_mag, scale)
+        mags_to_downsample = calculate_mags_to_downsample(from_mag, max_mag, scale)
+
+        if len(set([max(m.to_array()) for m in mags_to_downsample])) != len(
+            mags_to_downsample
+        ):
+            warnings.warn(
+                "The downsampling scheme contains multiple magnifications with the same maximum value. This is not supported by webknossos."
+            )
```

Comment:
When would this happen? Shouldn't we avoid this at all costs, since this will be very annoying for users.

Reply:
For example if the user uses `from_mag=(1, 1, 4)` with `CONSTANT_Z` (and some `max_mag`). This would lead to e.g. `1-1-4, 2-2-4, 4-4-4, 8-8-4, ...`.

Reply:
Another example (that does not require `CONSTANT_Z`):
`mode=ANISOTROPIC`, `from_mag=(2, 2, 1)`, `max_mag=(8, 8, 16)` and `scale=(2, 2, 1)`. This would result in `(4, 4, 2), (8, 8, 4), (8, 8, 8), (8, 8, 16)`.